Analyzer/src/TwoLepAnalyzer.cc

```diff
@@ -0,0 +1,442 @@
+                leptonScaleFactor = (GoodLeptons[0].first == "e") ? eleLepWeight : muLepWeight;
+            }
+            
+            //PileupWeight = tr.getVar<double>("_PUweightFactor");
+            bTagScaleFactor   = tr.getVar<double>("bTagSF_EventWeightSimple_Central");
+            htDerivedScaleFactor = tr.getVar<double>("htDerivedweight");
+            topPtScaleFactor = tr.getVar<double>("topPtScaleFactor");
+            prefiringScaleFactor = tr.getVar<double>("prefiringScaleFactor");
+            puScaleFactor = tr.getVar<double>("puWeightCorr");
+            
+            weight *= eventweight*leptonScaleFactor*bTagScaleFactor*htDerivedScaleFactor*prefiringScaleFactor*puScaleFactor;
+        }
+        // Create vector of only bjet TLorentzVectors
+        std::vector<TLorentzVector> GoodBJetsVec_pt30;
+        for (int j=0; j < Jets.size(); j++)
+        {
+            if (GoodBJets_pt30.at(j))
+            {
+                GoodBJetsVec_pt30.push_back(Jets.at(j));
+            }
+        }
+        // two lepton M(b,l) definition
+        TLorentzVector lep1;
+        TLorentzVector lep2;
+        int  used_jet;
+        double Mbl1;
+        double Mbl2;
+        double mbl1diff=999;
+        double mbl2diff=999;
+        if(NGoodLeptons==2)
+        {
+            if(GoodLeptons.at(0).second.Pt() >= GoodLeptons.at(1).second.Pt())
+            {
+                lep1 = GoodLeptons.at(0).second;
+                lep2 = GoodLeptons.at(1).second;
+            }
+            else
+            {
+                lep1 = GoodLeptons.at(1).second;
+                lep2 = GoodLeptons.at(0).second;
+            }
+            if(NGoodBJets_pt30 >= 2)
+            {
+                for(int b=0; b < NGoodBJets_pt30; b++)
+                {
+                    double  mbl1 = (lep1 + GoodBJetsVec_pt30.at(b)).M();
+                    if( abs(mbl1 - 105) < mbl1diff)
+                    {
+                        Mbl1 = mbl1;
+                        mbl1diff = abs(mbl1 - 105);
+                        used_jet = b;
+                    }
+                }
+                for(int b=0; b < NGoodBJets_pt30; b++)
+                {
+                    if(b != used_jet)
+                    {
+                        double mbl2 = (lep2 + GoodBJetsVec_pt30.at(b)).M();
+                        if( abs(mbl2 - 105) < mbl2diff)
+                        {
+                            Mbl2 = mbl2;
+                            mbl2diff = abs(mbl2 - 105);
+                        }
+                    }
+                }
+            }
+            else if (NGoodBJets_pt30 == 1)
+            {
+                if ( abs((lep1 + GoodBJetsVec_pt30.at(0)).M() - 110) <= abs((lep2 + GoodBJetsVec_pt30.at(0)).M() - 110) )
+                {
+                    Mbl1 = (lep1 + GoodBJetsVec_pt30.at(0)).M();
+                    for (int j = 0; j < Jets.size(); j++)
+                    {
+                        double mbl2 = (lep2 + Jets.at(j)).M();
```

Comment:
You need to use the GoodJets_pt30 vector of bools. Right now this is using all jets including bad ones that we throw out.